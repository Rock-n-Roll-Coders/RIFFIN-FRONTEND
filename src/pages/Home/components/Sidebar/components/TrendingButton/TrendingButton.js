// Components / hooks
import { useNavigate, useLocation } from "react-router-dom";
import { useEffect, useState } from "react";

// MUI
import Button from '@mui/material/Button';
import TrendingUpIcon from '@mui/icons-material/TrendingUp';

const TrendingButton = () => {
  const [variant, setVariant] = useState("text")
  const navigate = useNavigate()
  const location = useLocation()
  const handleClick = () => navigate('/trending')

  useEffect(()=> {
    if(location.pathname.startsWith("/trending")) {
      setVariant("contained")
    } else {
      setVariant("text")
    }
  }, [location])

  return (  
    <Button 
      startIcon={<TrendingUpIcon />}
      onClick={handleClick}
      sx={{justifyContent: 'left'}}
      disableElevation
      variant={variant}
    >
      Trending
    </Button>  
  );
}
 
export default TrendingButton;