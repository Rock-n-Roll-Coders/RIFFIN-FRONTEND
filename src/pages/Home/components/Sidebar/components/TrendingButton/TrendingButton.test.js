import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import TrendingButton from './TrendingButton';
import {BrowserRouter, MemoryRouter} from 'react-router-dom'
import userEvent from "@testing-library/user-event";

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate
}));

test('variant is "text" when not viewing trending', ()=> {
  const component = (
    <MemoryRouter initialEntries={['/latest']}>
      <TrendingButton />
    </MemoryRouter>
  )
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

test('variant changes to "contained" when viewing trending', () => {

  const component = (
    <MemoryRouter initialEntries={['/trending']}>
      <TrendingButton />
    </MemoryRouter>
  )

  const tree = renderer
  .create(component)
  .toJSON();
  expect(tree).toMatchSnapshot();
})

test('routes to trending on click', async ()=> {
  const component = (
    <BrowserRouter>
      <TrendingButton />
    </BrowserRouter>
  )

  render(component)
  const user = userEvent.setup()
  await user.click(screen.getByRole('button'))
  expect(mockedNavigate).toHaveBeenCalledWith('/trending');
})