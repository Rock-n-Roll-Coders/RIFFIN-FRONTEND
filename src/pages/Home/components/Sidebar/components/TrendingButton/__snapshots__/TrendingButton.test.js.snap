// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variant changes to "contained" when viewing trending 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-l5i4be-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="TrendingUpIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
      />
    </svg>
  </span>
  Trending
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`variant is "text" when not viewing trending 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-13sqc32-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="TrendingUpIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
      />
    </svg>
  </span>
  Trending
</button>
`;
